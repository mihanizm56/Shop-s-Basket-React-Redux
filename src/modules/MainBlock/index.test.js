import React from "react"
import { shallow } from "enzyme"
import { MainBlock } from "./index"

describe("MainBlock test", () => {
  it("Renders correct default props", () => {
    const component = shallow(<MainBlock />)

    expect(component).toMatchSnapshot()
  })

  it("Renders correct normal props", () => {
    const component = shallow(
      <MainBlock
        userItems={[{id:1}]}
        increment={() => "test"}
        decrement={() => "test"}
        deleteItem={() => "test"}
        promocode={"test_code"}
        discount={false}
        checkDiscount={() => "test"}
        discountValue={5500}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
