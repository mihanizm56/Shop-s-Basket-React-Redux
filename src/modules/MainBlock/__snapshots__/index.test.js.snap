// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBlock test Renders correct default props 1`] = `
<styled.div>
  <styled.h2>
    Ваша корзина
  </styled.h2>
  <styled.div>
    <styled.div>
      <styled.p>
        Товар
      </styled.p>
      <styled.p>
        Описание
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p>
        Количество
      </styled.p>
      <styled.p>
        Цена
      </styled.p>
      <styled.p>
        Удалить
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div />
  <DiscountWrapper
    checkDiscount={[Function]}
    discount={false}
    discountValue={1800}
    promocode=""
    userItems={Array []}
  />
</styled.div>
`;

exports[`MainBlock test Renders correct normal props 1`] = `
<styled.div>
  <styled.h2>
    Ваша корзина
  </styled.h2>
  <styled.div>
    <styled.div>
      <styled.p>
        Товар
      </styled.p>
      <styled.p>
        Описание
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p>
        Количество
      </styled.p>
      <styled.p>
        Цена
      </styled.p>
      <styled.p>
        Удалить
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      key="1"
    >
      <styled.div
        style={
          Object {
            "backgroundImage": "url(pictureIsNotFound)",
          }
        }
      />
      <styled.div>
        <styled.p />
        <styled.p>
          Код: 
        </styled.p>
        <styled.p>
          Размер: 
        </styled.p>
        <styled.p>
          Цвет: 
        </styled.p>
      </styled.div>
      <EditNumber
        decrement={[Function]}
        id={1}
        increment={[Function]}
        numberOfItems={0}
        userItems={
          Array [
            Object {
              "id": 1,
            },
          ]
        }
      />
      <styled.p>
         руб.
      </styled.p>
      <styled.div
        onClick={[Function]}
        style={
          Object {
            "backgroundImage": "url(remove.png)",
          }
        }
      />
    </styled.div>
  </styled.div>
  <DiscountWrapper
    checkDiscount={[Function]}
    discount={false}
    discountValue={5500}
    promocode="test_code"
    userItems={
      Array [
        Object {
          "id": 1,
        },
      ]
    }
  />
</styled.div>
`;
